extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampExtractorHelper.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    /**
+     * Fetch artist details from mobile endpoint.
+     * <a href=https://notabug.org/fynngodau/bandcampDirect/wiki/rewindBandcamp+%E2%80%93+Fetching+artist+details>
+     * More technical info.</a>
+     */
+    public static JsonObject getArtistDetails(String id) throws ParsingException {
+        try {
+            return
+                    JsonParser.object().from(
+                            NewPipe.getDownloader().post(
+                                    "https://bandcamp.com/api/mobile/22/band_details",
+                                    null,
+                                    JsonWriter.string()
+                                            .object()
+                                            .value("band_id", id)
+                                            .end()
+                                            .done()
+                                            .getBytes()
+                            ).responseBody()
+                    );
+        } catch (final IOException | ReCaptchaException | JsonParserException e) {
+            throw new ParsingException("Could not download band details", e);
+        }
+    }
+
+    /**
+     * @param id    The image ID
+     * @param album Whether this is the cover of an album
+     * @return Url of image with this ID in size 10 which is 1200x1200 (we could also choose size 0
+     * but we don't want something as large as 3460x3460 here, do we?)
+     */
+    public static String getImageUrl(final long id, final boolean album) {
+        return "https://f4.bcbits.com/img/" + (album ? 'a' : "") + id + "_10.jpg";
+    }
+
+    /**
+     * @return <code>true</code> if the given url looks like it comes from a bandcamp custom domain
+     * or if it comes from bandcamp.com itself
+     */
+    public static boolean isSupportedDomain(final String url) throws ParsingException {
+
+        // Accept all bandcamp.com URLs
+        if (url.toLowerCase().matches("https?://.+\\.bandcamp\\.com(/.*)?")) return true;
+
+        try {
+            // Accept all other URLs if they contain a <meta> tag that says they are generated by bandcamp
+            return Jsoup.parse(
+                    NewPipe.getDownloader().get(url).responseBody()
+            )
+                    .getElementsByAttributeValue("name", "generator")
+                    .attr("content").equals("Bandcamp");
+        } catch (IOException | ReCaptchaException e) {
+            throw new ParsingException("Could not determine whether URL is custom domain " +
+                    "(not available? network error?)");
+        }
+    }
+
+    static DateWrapper parseDate(final String textDate) throws ParsingException {
+        try {
+            final Date date = new SimpleDateFormat("dd MMM yyyy HH:mm:ss zzz", Locale.ENGLISH).parse(textDate);
+            final Calendar calendar = Calendar.getInstance();
+            calendar.setTime(date);
+            return new DateWrapper(calendar, false);
```

Comment:
```suggestion
            final OffsetDateTime offsetDateTime = OffsetDateTime.parse(textDate, DateTimeFormatter.ofPattern(""));
            return new DateWrapper(offsetDateTime, false);
```

Reply:
Good catch. But that won't accept `10 Mar 2019 23:00:42 GMT` or `16 May 2017 00:00:00 GMT`

Reply:
Setting the locale the `US` solves the parsing issue.
```suggestion
            final OffsetDateTime offsetDateTime = OffsetDateTime.parse(
                textDate, DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss zzz", Locale.US));
            return new DateWrapper(offsetDateTime, false);
```
But, now I get this: 
<details><summary><b>Crash log</b></summary><p>

```
Text '16 May 2017 00:00:00 GMT' could not be parsed: Unable to obtain OffsetDateTime from TemporalAccessor: {InstantSeconds=1494892800},ISO,GMT resolved to 2017-05-16T00:00 of type java.time.format.Parsed
java.time.format.DateTimeParseException: Text '16 May 2017 00:00:00 GMT' could not be parsed: Unable to obtain OffsetDateTime from TemporalAccessor: {InstantSeconds=1494892800},ISO,GMT resolved to 2017-05-16T00:00 of type java.time.format.Parsed
	at java.time.format.DateTimeFormatter.createError(DateTimeFormatter.java:1920)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1855)
	at java.time.OffsetDateTime.parse(OffsetDateTime.java:402)
	at org.schabi.newpipe.extractor.services.bandcamp.extractors.BandcampExtractorHelper.parseDate(BandcampExtractorHelper.java:152)
	at org.schabi.newpipe.extractor.services.bandcamp.extractors.BandcampStreamExtractor.getUploadDate(BandcampStreamExtractor.java:106)
	at org.schabi.newpipe.extractor.services.bandcamp.BandcampRadioStreamExtractorTest.testUploadDate(BandcampRadioStreamExtractorTest.java:88)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.time.DateTimeException: Unable to obtain OffsetDateTime from TemporalAccessor: {InstantSeconds=1494892800},ISO,GMT resolved to 2017-05-16T00:00 of type java.time.format.Parsed
	at java.time.OffsetDateTime.from(OffsetDateTime.java:370)
	at java.time.format.Parsed.query(Parsed.java:226)
	at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1851)
	... 55 more
Caused by: java.time.DateTimeException: Unable to obtain ZoneOffset from TemporalAccessor: {InstantSeconds=1494892800},ISO,GMT resolved to 2017-05-16T00:00 of type java.time.format.Parsed
	at java.time.ZoneOffset.from(ZoneOffset.java:348)
	at java.time.OffsetDateTime.from(OffsetDateTime.java:359)
	... 57 more
```

</details>

The ZoneOffset cannot be obtained. @Isira-Seneviratne Any idea?

Reply:
Using `ZonedDateTime` instead of `OffsetDateTime` works, and the locale doesn't need to be set to US.

The `ZonedDateTime` can be converted using `toOffsetDateTime()`.

Reply:
```suggestion
            final ZonedDateTime zonedDateTime = ZonedDateTime.parse(textDate, DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm:ss zzz"));
            return new DateWrapper(zonedDateTime.toOffsetDateTime(), false);
```

Reply:
Thanks for the help! 
Are you located in an English speaking country? When running the method without setting the locale to `ENGLISH` or any English-speaking country, the month cannot be parsed. I implemented your suggestions in 74b46fe

Reply:
Yeah, I'm in Sri Lanka.